lte/gateway/python/magma/mobilityd/ip_allocator_dhcp.py

```diff
@@ -133,7 +133,7 @@ def alloc_ip_address(self, sid: str, vlan_id: int) -> IPDesc:
             dhcp_desc = self._alloc_ip_address_from_dhcp(mac, vlan_id)
 
         if dhcp_desc and dhcp_allocated_ip(dhcp_desc):
```

Comment:
```suggestion
        if dhcp_desc and dhcp_desc.subnet and dhcp_allocated_ip(dhcp_desc)
```
Mypy is complaining that the input may be `None`. Converting to a string doesn't actually help in that case, it will also fail for the string "None".